### PR TITLE
adding RunIf

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?

Fixes https://github.com/Lightning-AI/utilities/issues/45#issuecomment-1449498765

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]
We need to install pytorch in the CI workflow 👀 Overall LGTM!

File: requirements/devel.txt [hunk 1]
```diff
@@ -3,3 +3,6 @@ codecov>=2.1
 pytest==7.2.0
 pytest-cov==4.0.0
 pytest-timeout==2.1.0
+
+#needed for development/testing
```
Comment (Borda) [inline requirements/devel.txt:7]:
```suggestion
# needed for development/testing
```

File: src/lightning_utilities/testing/runif.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            min_torch: Require that PyTorch is greater or equal than this version.
+            max_torch: Require that PyTorch is less than this version.
+            min_python: Require that Python is greater or equal than this version.
+            skip_windows: Skip for Windows platform.
+            standalone: Mark the test as standalone, our CI will run it in a separate process.
+            This requires that the ``RUN_STANDALONE_TESTS=1`` environment variable is set.
+            min_cuda_gpus: Require this number of gpus and that the ``PL_RUN_CUDA_TESTS=1`` environment variable is set.
+            is_mps_gpu: If True: Require that MPS (Apple Silicon) is available,
+            if False: Explicitly Require that MPS is not available
+            **kwargs: Any :class:`pytest.mark.skipif` keyword arguments.
```
Comment (Borda) [inline src/lightning_utilities/testing/runif.py:56]:
```suggestion
            **kwargs: Any :class:`pytest.mark.skipif` keyword arguments except `condition` and `reason`.
```

File: src/lightning_utilities/testing/runif.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        cls,
+        *args: Any,
+        min_torch: Optional[str] = None,
+        max_torch: Optional[str] = None,
+        min_python: Optional[str] = None,
+        skip_windows: bool = False,
+        standalone: bool = False,
+        min_cuda_gpus: int = 0,
+        is_mps_gpu: Optional[bool] = None,
+        **kwargs: Any,
```
Comment (carmocca) [inline src/lightning_utilities/testing/runif.py:41]: We've discussed this several times already :cry:
RunIf cannot be added as it is package-dependent. Only min/max_python and skip_windows could be

Comment (carmocca) [reply src/lightning_utilities/testing/runif.py:41]: Do you mean in lightning?

Comment (Borda) [reply src/lightning_utilities/testing/runif.py:41]: yes

Comment (Borda) [reply src/lightning_utilities/testing/runif.py:41]: yes

Comment (carmocca) [reply src/lightning_utilities/testing/runif.py:41]: I would suggest `utilities/testing.py`, but yes. That is a better way

Comment (Borda) [reply src/lightning_utilities/testing/runif.py:41]: well, it is already in utilities/testing/runif with optional import for all so zero base dependencies...

Comment (carmocca) [reply src/lightning_utilities/testing/runif.py:41]: It's not just about whether the dependency is installed or not. This torch-specific logic does not belong here. It's like adding a Trainer to torchmetrics

Comment (Borda) [reply src/lightning_utilities/testing/runif.py:41]: so then I'll need to create a new package `utilities-useful and use it in almost all projects

Comment (carmocca) [reply src/lightning_utilities/testing/runif.py:41]: I'm sorry but I don't understand. I said that `lightning/src/pytorch/utilities/testing.py` was fine. What is the issue with that?

**Pull request closed without merging**